### PR TITLE
Make string CommentLine reflect keyboard input

### LINKED ISSUE #539: Can't type operators in comments
Steps to recreate:
1. Type "#" to start writing a comment
2. Try to type "+" or "*" -- doesn't work, but typing ASCII characters does

### PR DESCRIPTION
**Pull request by zhixiangteoh**
Currently, CommentLine only allows `SChar` construction.

Goal of this PR is to make string CommentLine reflect exactly keyboard 
input, to achieve the following (as example):
- ``# Abl3 to +ype any symb0! 100.89*% ( [ ) ] \ = -_ '~"hello" ; let := >< ?? // $ ^& @hazel \` ``

Modify `JSUtil.is_single_key` to take in a variable `regex`, and then
within `CommentLine` match a wider `char_regex` (`^[^#]$` everything
except `#`) than `alpha_regex` (`^[a-zA-Z']$` all alphabets and `'`) on
regular lines.

Screen Shot 2021-08-09 at 10 57 41 AM

Currently still doesn't support alt/opt-modified symbols.

### CODE REVIEW

**Comment by cyrus-** [comment]
@zhixiangteoh status of this?

**Comment by zhixiangteoh** [comment]
alt/opt-modified keys “bug” seems most likely related to lack of ocaml support for unicode/utf8—e.g., typing `å` and logging `evt##.key` gives `"å”`, but logging `Key.get_key(evt)` (i.e., `Js.to_string(Js.Optdef(evt##.key))`) gives string as `Ã¥`

**Comment by cyrus-** [comment]
@zhixiangteoh OK, it's fine if it doesn't work for now -- we can fold that issue into future Unicode support for Hazel more broadly.

There are some conflicts with the dev branch -- can you resolve those, then mark as ready for review?

**Comment by cyrus-** [comment]
@zhixiangteoh is it ready for review now? If so, hit the "Ready for review" button

File: src/hazelcore/Action_Exp.re [hunk 1]
```diff
@@ -632,6 +632,11 @@ and syn_perform_block =
   /* Backspace & Delete */
   // Handle 2 special cases for CommentLines
   // Case 1
+  // e.g.,
+  //  #  comment 1
+  //  #| comment 2
```
Comment (cyrus-) [inline src/hazelcore/Action_Exp.re:637]: only seeing this behavior for blocks in synthetic position -- still has the previous behavior in analytic position

File: src/hazelweb/ModelAction.re [hunk 2]
```diff
@@ -47,4 +47,5 @@ type t =
   | ToggleHiddenHistoryAll
   | TogglePreviewOnHover
   | UpdateFontMetrics(FontMetrics.t)
-  | SerializeToConsole(serialize_object);
+  | SerializeToConsole(serialize_object)
+  | SerializeZexpToConsole;
```
Comment (cyrus-) [inline src/hazelweb/ModelAction.re]: instead of a new action, add ZExp as another serialize_object case

File: src/hazelweb/model/Program.re [hunk 3]
```diff
@@ -160,6 +160,36 @@ exception FailedAction;
 exception CursorEscaped;
 let perform_edit_action = (a, program) => {
   let edit_state = program.edit_state;
+  // switch (edit_state) {
```
Comment (cyrus-) [inline src/hazelweb/model/Program.re]: don't include big chunks of commented out code in PRs

**Review by cyrus- — CHANGES_REQUESTED** [review]
looking close! in addition to the comments here, needs to be brought up to date with dev.

**Comment by cyrus-** [comment]
@zhixiangteoh any updates here?

**Comment by zhixiangteoh** [comment]
> @zhixiangteoh any updates here?

Aim to merge dev and push by end of week.

**Review by cyrus- — APPROVED** [review]

**Pull request merged**